File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int n = nums.length;
        double sum = 0;
        for (int i = 0; i < k; i++) {
            sum += nums[i];
        }
        double avg = (sum / k);
        double max = avg;
        int i = 0;
        int j = k;
        while (j < n) {
            sum = sum - nums[i] + nums[j];
            avg = sum / k;
            max = Math.max(avg, max);
            i++;
            j++;
        }
        return max;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function (nums, k) {
    let maxAvg = -Infinity;
    for (let i = 0; i < nums.length - k + 1; i++) {
        let sum = 0;
        for (let j = i; j < k + i; j++) {
            sum += nums[j];
        }
        maxAvg = Math.max(maxAvg, sum/k);
    }
    return maxAvg;
};